### PR TITLE
01/10/24 default to text field if only custom button appears

### LINKED ISSUE #27: when only "custom" button should show, it should jsut show the field
IMAGE 2024-09-28 20:00:28

this is friction

### PR DESCRIPTION
**Pull request by hardingjam**
## Motivation

Some friction for users where there is a "Custom" button that must be clicked, despite no multi-choice option

## Solution

❗ First, merge #110 
Closes #27 

image

If there are no presets on a field, then show the text input directly

## Checks

By submitting this for review, I'm confirming I've done the following:
- [x] made this PR as small as possible
- [x] unit-tested any new functionality
- [x] linked any relevant issues or PRs

### CODE REVIEW

**Review by thedavidmeister — CHANGES_REQUESTED** [review]
no tests or screenshots for this on the PR

**Comment by hardingjam** [comment]
image

File: __mocks__/fixed-limit.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
+export const mockFixedLimit = `raindex-version: 8898591f3bcaa21dc91dc3b8584330fc405eadfa
```
Comment (thedavidmeister) [inline __mocks__/fixed-limit.ts:1]: why is this mock so large?

is all of this needed to test this test?

**Review by thedavidmeister — CHANGES_REQUESTED** [review]

File: __tests__/WebappFrame.test.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import WebappFrame from '@/app/_components/WebappFrame';
+import { Mock, vi } from 'vitest';
+import { generateButtonsData } from '@/app/_services/buttonsData';
+import { mockFixedLimit } from '@/__mocks__/fixed-limit';
```
Comment (thedavidmeister) [inline __tests__/WebappFrame.test.tsx:6]: fixtures?

**Review by thedavidmeister — DISMISSED** [review]

File: app/_components/WebappFrame.tsx [hunk 3]
```diff
@@ -170,6 +175,19 @@ const WebappFrame = ({ dotrainText, deploymentOption }: props) => {
 
 	const buttonsData = generateButtonsData(yamlData, currentState);
 
+	useEffect(() => {
+		console.log(buttonsData);
```
Comment (hardyjosh) [inline app/_components/WebappFrame.tsx:179]: console log

**Review by hardyjosh — APPROVED** [review]

**Pull request merged**